### ISSUE TITLE
Training code

### ISSUE THREAD

**Issue by buaa-luzhi** [laibe/M2U-Net#1]
@laibe 
Hello,  this is an excellent article.
When will the training code be made public?
Thank you!

**Comment by laibe** [comment]
@buaa-luzhi thank you. I'm currently working on a larger framework for retinal vessel segmentation that will include training scripts and the M2U-Net which I'm planning to make public by the end of summer (as a rough estimate).

**Comment by deepseek** [comment]
@laibe Thank you for your hard work.

Will your training scripts allow us to train using our own custom data? I am asking because in your published paper, you mainly state focusing on three datasets: Drive, Chase & HRF.
So, it would be amazing if we can train with a simple directory structure of:
- Input original images
- Input mask images

Also, will your training script allow training images that are different in size than each other?

**Comment by deepseek** [comment]
@laibe any updates on your training code release date?

**Comment by deepseek** [comment]
@laibe Any updates?

**Comment by laibe** [comment]
Hi, please see https://gitlab.idiap.ch/bob/bob.ip.binseg and read the docs on how to train the network.
